src/web/routes/figma/figma-router.ts

```diff
@@ -42,8 +43,25 @@ figmaRouter.post(
 
 				handleFigmaFileUpdateEventUseCase
```

Comment:
**suggestion:** Consider wrapping the code for asynchronous execution with [setImmediate](https://nodejs.org/en/learn/asynchronous-work/understanding-setimmediate) to make the intention explicit and the code self-descriptive.

```ts
figmaRouter.post(
	'/webhook',
	requestSchemaValidationMiddleware(FIGMA_WEBHOOK_EVENT_REQUEST_SCHEMA),
	figmaWebhookAuthMiddleware,
	(req: FigmaWebhookEventRequest, res: FigmaWebhookEventResponse) => {
		const webhookId = req.body.webhook_id;
		const { figmaTeam } = res.locals;

		switch (req.body.event_type) {
			case 'FILE_UPDATE': {
				const { file_key } = req.body;

				setImmediate(async () => {
					try {
						await handleFigmaFileUpdateEventUseCase.execute(
							figmaTeam,
							file_key,
						);
						getLogger().info(
							{ webhookId },
							'Figma webhook callback succeeded.',
						);
					} catch (e) {
						getLogger().error(e, 'Figma webhook callback failed.', {
							webhookId,
						});
					}
 
        	// Since Jest does not have the ability to wait on any pending
				  // promises, we need to manage this ourselves to signal that the
				  // async code has finished running
				  if (process.env.NODE_ENV === 'test') {
           completePendingRouteExecutionForTests(); 
          }
				});

				// Immediately send a 200 back to Figma, before doing any of our own
				// async processing
				return res.sendStatus(HttpStatusCode.Ok);
			}
			default:
				return res.sendStatus(HttpStatusCode.Ok);
		}
	},
);
```

We could go even further and move the asynchronous code to `src/jobs/handle-figma-file-update-event.ts`. This will help to simplify a route handler and get the code ready for the long-term queue-based solution (as we will just need to call the job in the new code).  

```ts
figmaRouter.post(
	'/webhook',
	requestSchemaValidationMiddleware(FIGMA_WEBHOOK_EVENT_REQUEST_SCHEMA),
	figmaWebhookAuthMiddleware,
	(req: FigmaWebhookEventRequest, res: FigmaWebhookEventResponse) => {
		const webhookId = req.body.webhook_id;
		const { figmaTeam } = res.locals;

		switch (req.body.event_type) {
			case 'FILE_UPDATE': {
				const { file_key } = req.body;

				setImmediate(() => void handleFigmaFileUpdateEvent(req.body, figmaTeam));

				// Immediately send a 200 back to Figma, before doing any of our own
				// async processing
				return res.sendStatus(HttpStatusCode.Ok);
			}
			default:
				return res.sendStatus(HttpStatusCode.Ok);
		}
	},
);
```